app/src/openxr/cpp/DeviceDelegateOpenXR.cpp

```diff
@@ -463,6 +464,19 @@ struct DeviceDelegateOpenXR::State {
       sessionBeginInfo.primaryViewConfigurationType = viewConfigType;
       CHECK_XRCMD(xrBeginSession(session, &sessionBeginInfo));
       vrReady = true;
+
+      // If hand tracking is supported, we want to emulate a default interaction
+      // profile, so that if Wolvic is launched without controllers active, we can
+      // still use hand tracking for emulating the controllers.
+      // This is a temporary situation while we don't implement WebXR hand tracking
+      // APIs.
+      if (mHandTrackingSupported && controllersReadyCallback != nullptr && input) {
+        input->UpdateInteractionProfile(*controller, OculusTouch.path);
+        if (input->AreControllersReady()) {
+           controllersReadyCallback();
+           controllersReadyCallback = nullptr;
+        }
+      }
```

Comment:
This is exactly the same of what we do in response to `XR_TYPE_EVENT_DATA_INTERACTION_PROFILE_CHANGED` so it'd be better to add this code to a new method in `DeviceDelegateOpenXR::State` and call it from both locations.

Reply:
Ok, will factorize that code in a new state method.